packages/kokoas-client/src/pages/projInvoice/fieldComponents/selectInvoices/InvoicesDialogContent.tsx

```diff
@@ -1,75 +1,113 @@
-import { DialogContent, FormControlLabel, Radio, RadioGroup, Typography } from '@mui/material';
+import { Box, Button, DialogContent, Tooltip, Typography } from '@mui/material';
 import { Stack } from '@mui/system';
 import { GetInvoicesByCustGroupId, useInvoicesByCustGroupId } from 'kokoas-client/src/hooksQuery';
 import WarningAmberIcon from '@mui/icons-material/WarningAmber';
-import { ListItemInvoices } from './ListItemInvoices';
 import { TInvoiceStatus } from '../../form';
+import { ListItemInvoices } from './ListItemInvoices';
+import { pages } from 'kokoas-client/src/pages/Router';
+import { generateParams } from 'kokoas-client/src/helpers/url';
+import { useNavigate } from 'react-router-dom';
+import { ReactNode } from 'react';
+import { Caption } from 'kokoas-client/src/components';
 
 export interface InvoicesDialogContentProps {
-  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
+  onClose: () => void;
   custGroupId: string
-  value: string
 }
 
 export const InvoicesDialogContent = (props: InvoicesDialogContentProps) => {
-  const { onChange, custGroupId, value } = props;
+  const { onClose, custGroupId } = props;
+  const navigate = useNavigate();
 
   const { data } = useInvoicesByCustGroupId<GetInvoicesByCustGroupId>(custGroupId);
 
   const { records: recInvoices } = data || {};
 
 
+  const handleClick = (newInvoiceId: string) => {
+    navigate(`${pages.projInvoice}?${generateParams({ invoiceId: newInvoiceId })}`);
+    onClose();
+  };
+
+
   /**
    * 選択肢の生成
    */
-  const actualOptions: OptionNode[] = recInvoices
+  type InvoiceOption = {
+    value: string,
+    key: string,
+    component: ReactNode,
+    tipComponent: ReactNode,
+  };
+
+  const actualOptions: InvoiceOption[] = recInvoices
     ?.filter(({ invoiceStatus }) => {
       return invoiceStatus.value as TInvoiceStatus !== 'voided';
     })
-    .map<OptionNode>((rec) => {
-    const { uuid } = rec;
+    ?.map<InvoiceOption>((rec) => {
+    const {
+      uuid,
+      estimateLists,
+    } = rec;
+
+    const infoToolTip: ReactNode = (
+      <Box width={'30%'}>
+        {estimateLists.value.map(({ value }) => {
+          return (
+            <Caption
+              text={`見積もり枝番：${value.dataId.value} 請求金額：${value.amountPerContract.value}`}
+              key={`${value.dataId.value}tooltipInfo`}
+            />);
+        })}
+      </Box>);
 
     return {
       value: uuid.value,
       key: uuid.value,
       component: (<ListItemInvoices invoiceRecord={rec} />),
+      tipComponent: infoToolTip,
```

Comment:
コンポーネント内にコンポネントを変数に格納ものは少ないならいいと思いますが、多くなると、分解したほういいと思います。

理由：

1. 処理速度。一回目は map で　actualOptions　を定義して、return で、また mapで回す。コンポーネントはオブジェクトなので、より重くなります。
2. 可読性。コンポーネント内でのレンダリング処理は、コンポーネントの中で行われるため、コンポーネントの中身が複雑になってしまい、可読性が低下してしまうということが挙げられます。一方、戻り値でレンダリング処理を行う場合は、コンポーネントの中身がシンプルになり、コードの理解や修正が容易になります。
3. 汎用性。コンポーネント内でデータだけを扱い、レンダリング処理を戻り値で行う場合、同じデータに対して、複数のコンポーネントを使い回すことができるからです。つまり、同じデータを扱う場面が増えたとしても、そのデータに対するレンダリング処理は変わらず、再利用が容易になります。

以上の理由から、データのみでレンダリング処理は戻り値にすることが望ましい場合があると考えられます。

```suggestion
      ...
      record: rec,
      }
      
      ...
      
      return (
      ...
        <ListItemInvoices invoiceRecord={rec} />
      ...
      )
```

参考：https://github.com/alan2207/bulletproof-react/blob/master/docs/components-and-styling.md